sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorReceiver.java

```diff
@@ -99,19 +169,23 @@ protected ReactorReceiver(String entityPath, Receiver receiver, ReceiveLinkHandl
     }
 
     @Override
-    public void addCredits(int credits) {
-        if (!isDisposed.get()) {
+    public Mono<Void> addCredits(int credits) {
+        if (isDisposed()) {
+            return monoError(logger, new IllegalStateException("Cannot add credits to closed link: " + getLinkName()));
+        }
+
+        return Mono.create(sink -> {
             try {
-                dispatcher.invoke(() -> receiver.flow(credits));
+                dispatcher.invoke(() -> {
+                    receiver.flow(credits);
+                    sink.success();
+                });
             } catch (IOException e) {
-                logger.warning("Unable to schedule work to add more credits.", e);
+                sink.error(new RuntimeException(String.format(
```

Comment:
Use `UncheckedIOException` instead when mapping IOException to a runtime exception.
https://docs.oracle.com/javase/8/docs/api/java/io/UncheckedIOException.html

Reply:
Sweet. I've never heard of this.